### PR TITLE
Implementation of virtual heartbeat monitor to help detect network (#…

### PR DESCRIPTION
**Pull request by kensipe**
…4069) (#4123) (#4147)

partitions between mesos-master and marathon.
Conflicts:
	src/main/scala/mesosphere/marathon/MarathonConf.scala
	src/main/scala/mesosphere/marathon/MarathonModule.scala
	src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala
	src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala
Conflicts:
	src/main/scala/mesosphere/marathon/MarathonModule.scala
	src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala
	src/main/scala/mesosphere/marathon/SchedulerCallbacksServiceAdapter.scala
	src/main/scala/mesosphere/marathon/state/Group.scala
	src/test/scala/mesosphere/marathon/MarathonSchedulerServiceTest.scala

### CODE REVIEW

**Comment by kensipe** [comment]
@jdef @aquamatthias Looking for review

File: src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala [hunk 1]
```diff
@@ -417,4 +433,18 @@ class MarathonSchedulerService @Inject() (
       offerLeadership()
     }
   }
+
+  private def startLeaderDurationMetric() = {
```
Comment (jdef) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala]: why these two new funcs? I don't seem them called as part of this PR...

**Comment by kensipe** [comment]
@jdef I removed the extra methods...   local tests pass.

**Comment by jdef** [comment]
lgtm

**Pull request merged**